StarRailBattleSim/src/relics/ornament/CelestialDifferentiator.java

```diff
@@ -0,0 +1,42 @@
+package relics.ornament;
+
+import characters.AbstractCharacter;
+import enemies.AbstractEnemy;
+import powers.AbstractPower;
+import powers.PermPower;
+import powers.PowerStat;
+import relics.AbstractRelicSetBonus;
+
+import java.util.ArrayList;
+
+public class CelestialDifferentiator extends AbstractRelicSetBonus {
+    public CelestialDifferentiator(AbstractCharacter owner, boolean fullSet) {
+        super(owner, fullSet);
+    }
+
+    public CelestialDifferentiator(AbstractCharacter owner) {
+        super(owner);
+    }
+
+    @Override
+    public void onEquip() {
+        this.owner.addPower(PermPower.create(PowerStat.CRIT_DAMAGE, 16, "Celestial Differentiator CD bonus"));
+    }
+
+    @Override
+    public void onCombatStart() {
+        this.owner.addPower(new CelestialDifferentiatorCRBonus());
+    }
+
+    public static class CelestialDifferentiatorCRBonus extends AbstractPower {
```

Comment:
The CR boost is only for one attack, so it removes itself in afterAttackFinish. Does it need PermPower to not be removed on turn start?

Reply:
Mainly for the edge case that the character doesn't attack on turn 1. PermPower has `lastsForever` set to true, if a power doesn't have that set, it will be removed at end of turn if its turn duration is 0 or less, and since the default turn duration is 0, this power would be automatically removed at the end of the character's first turn if it isn't used.

Generally, if a power manually removes itself instead of being turn-based, it should extend PermPower or set lastsForever to true so it can properly manage its own removal without interference.